### ISSUE TITLE
AGPL-3.0-only or AGPL-3.0-or-later?

### ISSUE THREAD

**Issue by yu-re-ka** [drakkan/sftpgo#891]
When looking at [the SPDX license list](https://spdx.org/licenses/), there is not a "AGPL-3.0" entry. This is because the author has to specify if users can choose to license the program under a newer version:

> Each version is given a distinguishing version number.  If the
Program specifies that a certain numbered version of the GNU Affero General
Public License "or any later version" applies to it, you have the
option of following the terms and conditions either of that numbered
version or of any later version published by the Free Software
Foundation.

https://spdx.org/licenses/AGPL-3.0-only.html
https://spdx.org/licenses/AGPL-3.0-or-later.html

Which of these applies to sftpgo?

**Comment by drakkan** [comment]
Hi,

I am not a lawyer or licensing expert. I just added the license from the GitHub [template](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/licensing-a-repository). 

When I submitted the homebrew package I was asked to choose between AGPL-3.0 only and AGPL-3.0 or later. I used [AGPL-3.0 only](https://formulae.brew.sh/formula/sftpgo). If a new AGPL license will be released I will consider switching to the new version after reading it.

**Comment by yu-re-ka** [comment]
Thanks! This answers my question. As a bonus, the README section could be updated for anyone who has the same question.
There is an example section inside the license itself for how to state that you are using the license, and this also includes these versioning terms.

https://github.com/drakkan/sftpgo/blob/main/LICENSE#L621
> How to Apply These Terms to Your New Programs

(adapted for sftpgo, and version 3 only)
```
    Copyright (C) 2022 sftpgo contributors

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU Affero General Public License as published
    by the Free Software Foundation, version 3 of the License.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU Affero General Public License for more details.

    You should have received a copy of the GNU Affero General Public License
    along with this program.  If not, see <https://www.gnu.org/licenses/>.

**Comment by drakkan** [comment]
I'm not sure. Bigger projects than SFTPGo haven't changed those lines

https://github.com/minio/minio/blob/master/LICENSE
https://github.com/nextcloud/server/blob/master/COPYING

As stated before I'm not a lawyer or licensing expert. I'm not sure if the copyright notice should include only my name or something more generic like (does any contributor automatically become the copyright holder?)

> Copyright (C) 2022 Nicola Murino and other SFTPGo contributors

and the year maybe must be 2019 (when I first developed and published SFTPGo). 

I am hesitant to modify the license file. I'll add GNU AGPLv3-only in the main readme

**Comment by yu-re-ka** [comment]
Of course the license text should not be changed, however the license text suggests you adapt and copy over those lines to the source files to declare them as AGPL-licensed.

**Comment by drakkan** [comment]
> Of course the license text should not be changed, however the license text suggests you adapt and copy over those lines to the source files to declare them as AGPL-licensed.

Yes, but it is easy to do mistakes, for example if I add something like

> Copyright (C) 2019 Nicola Murino and other SFTPGo contributors

I'm assigning the copyright to any contributors. I don't know if this is right. There is no company behind SFTPGo so I cannot write "SFTPGo inc" or something similar. If I add only my name it could be not right for contributors (even if I wrote 99.9% of the SFTPGo code). I think I'll simply replace "GNU AGPLv3" with "GNU AGPL-3.0-only" [here](https://github.com/drakkan/sftpgo/blob/main/README.md#license)

**Comment by yu-re-ka** [comment]
Fine for me 👍🏻
